bftengine/src/bcstatetransfer/SourceSelector.hpp

```diff
@@ -105,6 +121,20 @@ class SourceSelector {
 
   uint16_t minPrePrepareMsgsForPrimaryAwarness() { return minPrePrepareMsgsForPrimaryAwarness_; }
 
+  // Metric
+  inline void updatePreferredReplicasMetric() const {
```

Comment:
please remove all inline keywards you've added here. It's not really necessary, You can see in the c++ standard:
`A function defined within a class definition is an inline function.`
This kind of definition is called "inline member function" so the compiler doesn't need the added keyword, which anyway is just a hint.